### PR TITLE
Add the kubediscovery JWS API as a standalone pod.

### PR DESCRIPTION
**Pull request by dgoodwin**
I managed to integrate the kubediscovery code into the main source tree and get it building, lots of notes are in the discovery/README.md.

I have a few files floating around that directory we'll need to get rid of as this progresses.

Tomorrow I can work on integrating this into kubeadm's manifests.go, or if you would like to take this over I can continue working on the server itself. I have some sample yaml files in discovery/ dir for how I have been deploying it as a replica set and with an associated secret for the CA cert.

Code is rough around the edges, still banging it into shape. Will be in touch for how we can divide work tomorrow.

Cheers!

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -363,7 +363,7 @@ function start_kubelet {
       if [[ -n "${NET_PLUGIN}" ]]; then
         net_plugin_args="--network-plugin=${NET_PLUGIN}"
       fi
-      
+
```
Comment (errordeveloper) [inline hack/local-up-cluster.sh:366]: could you please drop this, as it's affects rebases and isn't essential here...

File: cmd/kubediscovery/kubediscovery.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"log"
+	"net/http"
+	"os"
+
+	kd "k8s.io/kubernetes/pkg/kubediscovery"
+)
+
+func main() {
+
```
Comment (errordeveloper) [inline cmd/kubediscovery/kubediscovery.go:24]: I'm rather glad it hasn't got any flags, it probably isn't going to need them.

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (dgoodwin) [reply hack/local-up-cluster.sh:366]: Will do, I need a modification to this file to build and I keep accidentally committing parts of it.

**Comment by dgoodwin** [comment]
Closing for now per our discussion on slack, plan is to keep server commits separate and isolated for now to avoid git squash and authorship issues in future. This server will be published on dockerhub as dgoodwin/kubediscovery for time being, kubeadm will be modified to deploy this container as a pod.

In future we'll squash all our work and combine patches, and work on merging the server into mainline API.

**Pull request closed without merging**